src/index.js

```diff
@@ -92,7 +80,65 @@ export default class AutoCheckElement extends HTMLElement {
   }
 }
 
-function check(autoCheckElement: AutoCheckElement) {
+function makeDeferred<T>(): [Promise<T>, (T) => Promise<T>, (Error) => T] {
+  let resolve
+  let reject
+  const promise = new Promise(function(_resolve, _reject) {
+    resolve = _resolve
+    reject = _reject
+  })
+
+  if (!resolve) throw new Error('invariant: resolve')
+  if (!reject) throw new Error('invariant: reject')
+
+  return [promise, resolve, reject]
+}
+
+function makeAbortController() {
+  if ('AbortController' in window) {
+    return new AbortController()
+  }
+  return {signal: null, abort() {}}
```

Comment:
It's relatively simple to make a fully functional AbortController if one isn't present:

```suggestion
  let aborted = false
  return {signal: { get aborted() { return aborted }, abort() { aborted = true }}
```

Perhaps we can move that into fetch-helpers so we have a way to get proper abort controllers?